src/AWSCredentials.jl

```diff
@@ -56,392 +60,497 @@ mutable struct AWSCredentials
-    end
-end
-import Base: copy!
-Base.@deprecate copy!(dest::AWSCredentials, src::AWSCredentials) copyto!(dest, src)
-
 
 """
-Is Julia running in an AWS Lambda sandbox?
-"""
-localhost_is_lambda() = haskey(ENV, "LAMBDA_TASK_ROOT")
+    localhost_is_ec2()
 
+Checks to see if the localhost is on EC2.
 
-"""
-Is Julia running on an EC2 virtual machine?
+# Returns
+- `Bool`: True if running the machine executing this is an EC2 instance
 """
 function localhost_is_ec2()
-
-    if localhost_is_lambda()
-        return false
+    # Checking to see if you are running on an EC2 instance is a complicated problem due to
+    # a large amount of caveats. Below is a list of methods to implement to work through
+    # most of these problems:
+    #
+    # 1. Check the `hostname -d`; this will not work if using non-Amazon DNS
+    # 2. Check metadata with EC2 internal domain name `curl -s
+    # http://instance-data.ec2.internal`; this will not work with a VPC (legacy EC2 only)
+    # 3. Check `sudo dmidecode -s bios-version`; this requires `dmidecode` on the instance
+    # 4. Check `/sys/devices/virtual/dmi/id/bios_version`; this may not work depending on
+    # the instance, Amazon does not document this file however so it's quite unreliable
+    # 5. Check `http://169.254.169.254`; This is a link-local address for metadata,
+    # apparently other cloud providers make this metadata URL available now as well so it's
+    # not guaranteed that you're on an EC2 instance
+    # 6. When checking the UUID, check for little-endian representation
+
+    # This is not guarenteed to work on Windows as RNG can make the UUID begin with EC2 on a
+    # non-EC2 instance
+    if @mock Sys.iswindows()
+        command = `wmic path win32_computersystemproduct get uuid`
+        result = @mock Base.read(command, String)
+        instance_uuid = strip(split(result, "\n")[2])
+
+        if instance_uuid[1:3] == "EC2"
+            return true
+        else
+            return false
+        end
     end
 
-    if isfile("/sys/hypervisor/uuid") &&
-       String(read("/sys/hypervisor/uuid",3)) == "ec2"
+    # Note: This will not work on new m5 and c5 instances because they use a new hypervisor
+    # stack and the kernel does not create files in sysfs
+    if isfile("/sys/hypervisor/uuid") && String(read("/sys/hypervisor/uuid", 3)) == "ec2"
         return true
     end
 
     if isfile("/sys/devices/virtual/dmi/id/product_uuid")
         try
+            if read("/sys/devices/virtual/dmi/id/product_uuid", String) == "EC2"
+                return true
+            end
+        catch e
             # product_uuid is not world readable!
             # https://patchwork.kernel.org/patch/6461521/
             # https://github.com/JuliaCloud/AWSCore.jl/issues/24
-            if String(read("/sys/devices/virtual/dmi/id/product_uuid")) == "EC2"
-                return true
+            if e isa LoadError
+                return false
             end
```

Comment:
Should you `rethrow` in the case of another exception? A `LoadError` doesn't seem like the right exception to handle. Based upon the comment maybe you can avoid exception handling using `isreadable`